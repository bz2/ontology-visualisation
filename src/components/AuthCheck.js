import { useEffect } from 'react'
import { useRouter } from 'next/router'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import checkAuthAtStartup from '../utils/auth/checkTokenValidity'
import actions from '../store/actions'
import startupActions from '../utils/graphVisualisation/startupActions'
import { ROUTE_LOGIN } from '../constants/routes'
import setPageView from '../utils/analytics/setPageView'

const AuthCheck = ({
  updateStoreValue,
  user,
  showTour,
  pageProps
}) => {
  const { t } = useTranslation()

  const router = useRouter()

  const handleRouteChange = (url) => setPageView({ url, updateStoreValue })

  // check if authenticated, otherwise redirect to login
  useEffect(() => {
    if (!user.isGuest && user.email === '') {
      checkAuthAtStartup({
        router,
        updateStoreValue
      })
    }

    router.events.on('routeChangeComplete', handleRouteChange)

    return () => router.events.off('routeChangeComplete', handleRouteChange)
  }, [])

  useEffect(() => {
    if (user.email || user.isGuest) {
      startupActions({
        showTour,
        updateStoreValue,
        t
      })
    }
  }, [user])

  useEffect(() => {
    if (pageProps.statusCode === 404) {
      router.push(ROUTE_LOGIN)
    }
  }, [pageProps.statusCode])

  return null
}

AuthCheck.propTypes = {
  updateStoreValue: PropTypes.func.isRequired,
  user: PropTypes.shape().isRequired,
  pageProps: PropTypes.shape().isRequired,
}

const mapPropsToState = ({
  user,
  showTour
}) => ({
  user,
  showTour
})

export default connect(
  mapPropsToState,
  actions
)(AuthCheck)
