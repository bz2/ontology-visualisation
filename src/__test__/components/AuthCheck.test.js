import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AuthCheck from '../../components/AuthCheck'

const setup = () => {
  const props = {
    updateStoreValue: jest.fn(),
    user: {
      email: 'valid@email.coma'
    },
    pageProps: {}
  }

  const component = shallow(<AuthCheck {...props} />)

  return {
    component,
    props
  }
}

describe('AuthCheck', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const {
      component
    } = setup({
      isAuthCheckInternal: false
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
