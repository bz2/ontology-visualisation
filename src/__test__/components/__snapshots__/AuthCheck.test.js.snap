// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthCheck should match snapshot 1`] = `
<Connect
  actions={
    Object {
      "updateStoreValue": [Function],
    }
  }
  mapToProps={[Function]}
  pageProps={Object {}}
  updateStoreValue={[MockFunction]}
  user={
    Object {
      "email": "valid@email.coma",
    }
  }
>
  <Component />
</Connect>
`;
